src/algorithm.c

```diff
@@ -61,22 +61,27 @@ validate_certificate_public_key_algorithm(X509_ALGOR *pa)
 int
 validate_certificate_public_key_algorithm_bgpsec(X509_ALGOR *pa)
 {
+	const ASN1_OBJECT *obj;
+	int parameter_type;
+	const void *parameter;
 	int nid;
 
-	nid = OBJ_obj2nid(pa->algorithm);
+	X509_ALGOR_get0(&obj, &parameter_type, &parameter, pa);
+
+	nid = OBJ_obj2nid(obj);
 
 	/* Validate algorithm and parameters (RFC 8608#section-3.1.1) */
 	if (nid != NID_X9_62_id_ecPublicKey)
 		return pr_val_err("Certificate's public key format is NID '%s', not id-ecPublicKey.",
 		    OBJ_nid2sn(nid));
 
-	if (pa->parameter == NULL)
+	if (parameter == NULL)
```

Comment:
Issue: `X509_ALGOR_get0()` only sets `parameter` [if `pa->parameter` is not `NULL`](https://github.com/openssl/openssl/blob/openssl-3.2.0/crypto/asn1/x_algor.c#L80). FORT doesn't control `pa->parameter`, so this `if` might (theoretically) read uninitialized memory.

Possible solutions: Remove this `if`, swap it with the one that immediately follows, or pre-initialize `parameter` to `NULL`.

Reply:
Thanks. This is arguably a bug in the API. I forgot that I had fixed that in LibreSSL. I initialized parameters to NULL to preserve what was there.